internal/shim-sev/src/syscall.rs

```diff
@@ -696,6 +697,37 @@ impl Handler {
 
         Ok(0)
     }
+
+    pub fn poll(&self) -> Result<usize, libc::c_int> {
+        let nfds = self.b as libc::nfds_t;
+        let timeout = self.c as libc::c_int;
+        let trusted =
+            unsafe { core::slice::from_raw_parts_mut(self.a as *mut libc::pollfd, nfds as _) };
```

Comment:
This could be cleaned up by making the registers into `Register` types. But this would effect all the syscalls. Separate PR please.